docs/docfiles/user_guides/manual/dpctl/device_selection.rst

```diff
@@ -0,0 +1,143 @@
+runtime.
+
+The example :ref:`fig-gpu-device-selection` shows the usage of the
+:func:`dpctl.select_gpu_device()` device selection function. In case multiple
+GPU devices are available, only one is returned based on the underlying scoring
+logic inside the SYCL runtime. If the selection function was unable to select a
+device a ``ValueError`` is raised.
+
+.. _fig-gpu-device-selection:
+
+.. literalinclude:: ../../../../../examples/python/device_selection.py
+    :language: python
+    :lines: 20-21, 38-52
+    :caption: Selecting a GPU Device
+    :linenos:
+
+A possible output for the example :ref:`fig-gpu-device-selection` may be:
+
+.. program-output:: python ../examples/python/device_selection.py -r create_gpu_device
+
+Selecting a Device Using a Filter String
+----------------------------------------
+
+Along with using the default device selection functions, a more explicit way of
+device selection involves the use of *filter strings* (refer
+`oneAPI filter selection extension`_). The example
+:ref:`fig-gpu-device-selection` also demonstrates the use of a filter string
+to create a GPU device directly. Using a filter string allows much more
+fine-grained control for selecting a device. The following example
+:ref:`fig-filter-selection` demonstrates usages of device selection using filter
+strings.
+
+.. _fig-filter-selection:
+
+.. literalinclude:: ../../../../../examples/python/filter_selection.py
+    :language: python
+    :lines: 20-21, 23-53
+    :caption: Device Creation With Filter Strings
+    :linenos:
+
+A possible output for the example :ref:`fig-filter-selection` may be:
+
+.. program-output:: python ../examples/python/filter_selection.py -r select_using_filter
+
+
+It is also possible to pass a list of devices using a filter string. The
+example :ref:`fig-adv-device-selection` demonstrates such a use case. The
+filter string ``gpu,cpu`` implies that a GPU should be selected if available,
+else a CPU device should be selected.
+
+.. _fig-adv-device-selection:
+
+.. literalinclude:: ../../../../../examples/python/device_selection.py
+    :language: python
+    :lines: 20-21, 55-67
+    :caption: Selecting a GPU Device if Available
+    :linenos:
+
+A possible output for the example :ref:`fig-adv-device-selection` may be:
+
+.. program-output:: python ../examples/python/device_selection.py -r create_gpu_device_if_present
+
+.. Note::
+    A **filter string** is a three-tuple that may specify the *backend*,
+    *device type*, and *device number* as a colon (:) separated string. The
+    backend specifies the type of device driver and can have a value such as
+    *opencl*, *level-zero*, or *cuda*. The device type can be *gpu*, *cpu*,
+    *accelerator*. And, the device number is a numeric value specifying the
+    ordinality of the device in the listing of devices as determined by the SYCL
+    runtime. The filter values are dependent of what the underlying SYCL runtime
```

Comment:
We should state that the listing of devices given by the SYCL runtime is stable (same one seen by different processes, or from run to run, provided that driver configuration and `SYCL_DEVICE_FILTER` setting are the same.) In other words, `dpctl.SyclDevice("4")` will select the same device as using `sycl::ext::oneapi::filter_selector("4")` in a C++ application.

Reply:
Done. Please review again.